core/src/main/java/com/linecorp/armeria/server/logging/structured/ServiceLogEntryBuilder.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.ResponseLog;
+
+@FunctionalInterface
+public interface ServiceLogEntryBuilder<E> {
+    @Nullable
+    E build(RequestContext reqCtx, ResponseLog resLog);
```

Comment:
I just try to avoid using `java.util.function.*` interfaces as an argument or return value of public interface as I think it makes interface harder to read and understand.
Anyway it's just my personal preference. Let me drop this and use `BiFunction` instead if its better practice here.

> Question, when would this method return null?

When user-implemented service log builder thinks the context isn't proper to issue a service log entry?(i.e, server failed to parse thrift argument and returned bad request w/o creating even ThriftCall, user might don't want to issue an entry)